config/routes.rb

```diff
@@ -279,6 +280,7 @@
       :constraints => { view: /moderate/ }
   get "/listings/:category/:slug/delete_confirm" => "listings#delete_confirm"
   delete "/listings/:category/:slug" => "listings#destroy"
+  patch "listing_endorsement/:id" => "listing_endorsements#update", :as => :approve_endorsement
```

Comment:
why not simply add `  resources :listing_endorsements, only: %i[create update]` up there?